libs/file_processing/file_processing_core.py

```diff
@@ -229,14 +236,28 @@ def upload_binified_data(binified_data, error_handler, survey_id_dict):
         older data to/from S3 for each chunk.
         Returns a set of concatenations that have succeeded and can be removed.
         Returns the number of failed FTPS so that we don't retry them.
+        Returns the earliest and latest time bins handled
         Raises any errors on the passed in ErrorHandler."""
     failed_ftps = set([])
     ftps_to_retire = set([])
     upload_these = []
+
+    # Track the earliest and latest time bins, to return them at the end of the function
+    earliest_time_bin = None
+    latest_time_bin = None
+
+    def update_earliest_and_latest_time_bins(new_time_bin: int):
+        nonlocal earliest_time_bin, latest_time_bin
+        if earliest_time_bin is None or new_time_bin < earliest_time_bin:
+            earliest_time_bin = new_time_bin
+        if latest_time_bin is None or new_time_bin > latest_time_bin:
+            latest_time_bin = new_time_bin
```

Comment:
:art: I don't think it's worth factoring this out into its own function, especially if you have to specify `nonlocal`. This makes it harder to understand for another reader. I'd just inline these 4 lines of logic.

Reply:
I'm okay with adding these inlined.

This `upload_binified_data` function is not my best work, it needs to be converted to a class of some kind so that all the stuff in play can be instantiated, accessed, and deleted(!) cleanly.  (The control flow using a try-except-else is particularly shit.  I tried disentangling it once, but like the dwarves in Moria, in my greed I delved too deep and was consumed by a balrog ... I'm not totally sure what the bridge of Khazad-dûm represents in this metaphor.)